File: pset1 C/mario/mario.c
#include <stdio.h>
#include <cs50.h>

////////////////////////////////////////////////////////////////////////////////
//
//  Function name: main()
//  Purpose: Handles the user input of the program
//
////////////////////////////////////////////////////////////////////////////////
int main(void)
{
    bool check = false;
    do
    {
        int val = get_int("Height: ");
        if (val > 0 && val < 9)
        {
            check = true;




////////////////////////////////////////////////////////////////////////////////
//
//  Pyramid output process based on user's input of 'val'
//
////////////////////////////////////////////////////////////////////////////////
            //Counts down each row
            for (int i = 1; i <= val; i++)
            {

                // Row number: i
                //Spaces required for row
                int spaces = val - i;

                //Hashes required for each row
                int hashes = val - spaces;

                //Counts through spaces and prints them
                for (int j = 1; j <= spaces; j++)
                {
                    printf(" ");

                }

                //Counts through the hashes and prints them
                for (int j = 1; j <= hashes; j++)
                {
                    printf("#");

                }

                //Outputs two space gap
                printf("  ");

                //Outputs the second side of the pyramid
                for (int j = 1; j <= hashes; j++)
                {
                    printf("#");

                }

                //Goes to next line for next row output
                printf("\n");

            }
////////////////////////////////////////////////////////////////////////////////





        }
    }
    while (check == false);
}

File: pset6 Python/mario/more/mario.py
#Import get_int function from cs50 lib
from cs50 import get_int

#Set a while loop condition
condition = True

#Continue to loop until condition met
while condition:
    #Get height input
    n = get_int("Height: ")
    
    #Do a check on input height
    if 1 <= n < 9:
        #Loop through the height value and produce the graphic
        for i in range(n):
            #Calculate the spaces and hashes required for graphic
            spaces = n - (i + 1)
            hashes = n - spaces
    
            #Output the graphics using calculated values
            for j in range(spaces):
                print(" ", end = "")
    
            for j in range(hashes):
                print("#", end = "")
    
            print("  ", end = "")
    
            for j in range(hashes):
                print("#", end = "")
    
            print("\n", end = "")
            
            #Set condition to false and end loop
            condition = False
    
    else:
        #If height check fails output message
        print("Pick a number in the range of 1 and 8")